### ISSUE TITLE
Improve token handling

### ISSUE THREAD

**Issue by dmeroff** [dektrium/yii2-user#371]
- [ ] Hash tokens in the database to improve security
- [ ] Remove user's id from token urls
- [ ] Fix url rules
- [ ] Remove old tokens when creating new tokens #340 
- [ ] Add console command to remove outdated tokens
- [ ] Ease the addition of new token types
- [ ] Tests

**Comment by thiagotalma** [comment]
@dmeroff In my specific case, I need to generate a token with 6 digits and send via SMS. Currently I do it well inelegant way.

You'd better think about creating a validation configurable way so that developers have more freedom without having to create monsters  :smiley:

**Comment by dmeroff** [comment]
@thiagotalma Yeah I think I can add a new service layer which can be overridden to introduce your own ways of confirmation. By default it will work only with email-based confirmation messages.

**Comment by thiagotalma** [comment]
:+1:
